src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
This is to remove one item in case we are full. I not familiar with the algebra of the `distance_exp(node_id)` but maybe there is a way to combine two of them `node_id ⨁ node_id -> node_id` that allows to calculate a desired "combined distance". Do you think that's possible?